src/listpack.c

```diff
@@ -698,6 +698,9 @@ unsigned char *lpFind(unsigned char *lp, unsigned char *p, unsigned char *s,
     while (p) {
         if (skipcnt == 0) {
             value = lpGetWithSize(p, &ll, NULL, &entry_size);
+            /* check the value doesn't reach outside the listpack before accessing it */
+            assert(p >= lp + LP_HDR_SIZE && p + entry_size < lp + lp_bytes);
```

Comment:
I think this can be moved into before `if (slen == ll && memcmp(value, s, slen) == 0)`.
When entry is a number, we do not read illegal memory.

Reply:
i was thinking to also cover `p += entry_size;`
but i suppose that's covered by the check we do after it..

another thought is that maybe it also asserts that the data we already read is valid before actually using it (i.e. the int value).

but i suppose that when viewed in the context of the whole function, the +8 at the bottom covers all of that..